src/front/glsl_new/ast.rs

```diff
@@ -49,7 +49,47 @@ pub enum Profile {
 pub struct Context {
     pub expressions: Arena<Expression>,
     pub local_variables: Arena<LocalVariable>,
-    pub lookup_local_variables: FastHashMap<String, Handle<LocalVariable>>,
+    pub scopes: Vec<FastHashMap<String, Handle<LocalVariable>>>,
+}
+
+impl Context {
+    pub fn lookup_local_var(&self, name: &str) -> Option<&Handle<LocalVariable>> {
```

Comment:
no need to return a reference to a handle, since the are copyable - just return the handle

Reply:
done